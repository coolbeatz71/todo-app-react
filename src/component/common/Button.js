import React from 'react';
import PropTypes from 'prop-types';

/**
 * Button component
 * @returns {void}
 * @param {object} props
 */
const Button = ({ children, onClick, type, disabled, myClass }) => (
  <button type={type} disabled={disabled} onClick={onClick} className={`button ${myClass}`}>
    {children}
  </button>
);

Button.propTypes = {
  type: PropTypes.string,
  children: PropTypes.any.isRequired,
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  myClass: PropTypes.string,
};

Button.defaultProps = {
  type: 'button',
  disabled: false,
  myClass: '',
};

export default Button;
