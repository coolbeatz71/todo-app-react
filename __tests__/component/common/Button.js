import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../../src/component/common/Button';

const props = {
  children: 'Testing',
};
describe('Button.js', () => {
  test('should render the button component', () => {
    const wrapper = shallow(<Button {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
