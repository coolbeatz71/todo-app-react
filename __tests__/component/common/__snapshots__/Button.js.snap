// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.js should render the button component 1`] = `ShallowWrapper {}`;
